tests/test_feedexport.py

```diff
@@ -1780,14 +1780,15 @@ class FileFeedStoragePreFeedOptionsTest(unittest.TestCase):
     maxDiff = None
 
     def test_init(self):
-        settings_dict = {
-            'FEED_URI': 'file:///tmp/foobar',
-            'FEED_STORAGES': {
-                'file': FileFeedStorageWithoutFeedOptions
-            },
-        }
-        crawler = get_crawler(settings_dict=settings_dict)
-        feed_exporter = FeedExporter.from_crawler(crawler)
+        with tempfile.NamedTemporaryFile() as temp:
+            settings_dict = {
+                'FEED_URI': f'file:///{temp.name}',
```

Comment:
What is the value of `temp.name` here? Aren’t there 1 too many / in the URI scheme here? And if so, how are tests working nonetheless?

Also, according to https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile:

> Whether the name can be used to open the file a second time, while the named temporary file is still open, varies across platforms (it can be so used on Unix; it cannot on Windows NT or later).

Which makes me wonder how tests are passing in Windows.

Reply:
@Gallaecio temp.name returns the random name of the file. We create a new file with a random name to resolve the permissions issue at each test run, and the test is passing successfully.

Reply:
`f'file:///{temp.name}'` generates a URI like `file:////tmp/tmp_4p8ym67`, which seems to work but has 1 too many slashes at the beginning. Replacing `f'file:///{temp.name}'` with `f'file://{temp.name}'` gives me peace of mind and still works.

As for the platform issue, I see that the issue described in the documentation affects the file while open; what we are doing here is create a random file with a file-system path, remove it (when leaving the context manager), and pass it’s path to the spider which later recreates the file. It feels a bit like a hack to me, since I am not aware of any assurances that, after the file is first removed, another call to `NamedTemporaryFile` will not create a new file on the same location, but there’s probably some randomness in the file name choosing algorithm that makes that scenario very unlikely, so :+1: from me.

Reply:
```suggestion
                'FEED_URI': f'file://{temp.name}',
```